lib/scholar/linear_regression.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Scholar.LinearRegression do
+
+  defstruct :coefficients
+
+  @doc """
+  Fits a linear regression model to the given dataset.
+  """
+  def fit(x, y) do
+    {coeff, intercept} = lstsq(x, y)
+  end
+
+  defnp lstsq(x, y) do
+    {u, s, vt} = Nx.LinAlg.svd(x)
+
+    uty =
+      u
+      |> Nx.transpose()
+      |> Nx.dot(y)
+
+    s_inv = Nx.new_axis(1 / s, -1)
+
+    x =
+      vt
+      |> Nx.transpose()
+      |> Nx.dot(Nx.multiply(s_inv, uty))
+
+    Nx.transpose(x)
+  end
+
+  defimpl Scholar do
+    import Nx.Defn
+
+    def predict(%LinearRegression{coefficients: coef}, x) do
```

Comment:
I think we can have `predict` actually defined in `Scholar.LinearRegression` and then we just delegate to it. We should probably define `Nx.Container` for the struct too?